File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    new_s = []
    answer = []
    s = list(s)

    for i in range(len(s)-1, -1, -1):
        cnt = 0

        for k in range(i-1, -1, -1):
            if (s[i] == s[k]):
                new_s.append(i-k)
                cnt += 1
                break
        if (cnt == 0):
            new_s.append(-1)

    for i in reversed(new_s):
        answer.append(i)

    return answer

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.HashMap;
import java.util.Map;

public class Solution {

    public static int[] solution(String s) {
        int[] answer = new int[s.length()];
        Map<String, Integer> wordInformation = new HashMap<>();
        char[] words = s.toCharArray();

        for (int i = 0; i < s.length(); i++) {
            String word = String.valueOf(words[i]);
            if (!wordInformation.containsKey(word)) {
                answer[i] = -1;
                wordInformation.put(word, i + 1);
            } else {
                int position = i + 1 - wordInformation.get(word);
                answer[i] = position;
                wordInformation.put(word, i + 1);
            }
        }

        return answer;
    }
}
